File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.cpp

class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode*curr= head;
        int length = 0;
        while(curr){
            length +=1;
            curr = curr->next;
        }
        if(length == n){
            return head->next;
        }
        ListNode*k = head;
        int i = 0;
        while(k){
            if((length-i)==n+1){
                k->next = k->next->next;
                break;
            }
            k=k->next;
            i+=1;
        }
        return head;
        
    }
};

File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:    
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        c=[]
        k = ListNode(0)
        curr = k

        for i in lists:
            while i:
                c.append(i.val)

                i = i.next


        c.sort()

        for n in c:
            curr.next = ListNode(n)
            curr= curr.next


        return k.next

        
        